File: 04_Data_Structures_&_Counting/14425_String_Set_Membership/Answer14425.ts
const [numArr, ...strs] = `${require('fs').readFileSync(0)}`.split('\n')
const [n, m] = numArr.split(' ').map(v=>+v)
const strsSet = new Set(strs.slice(0, n))
let resultInt= 0
for(let i= n; i < n + m; i++) if(strsSet.has(strs[i])) resultInt++
console.log(resultInt)

File: 04_Data_Structures_&_Counting/14425_String_Set_Membership/Answer14425.java
public class Answer14425 {
  public static void main(String[] args) {
    var s = new java.util.Scanner(System.in);
    int n = s.nextInt();
    int m = s.nextInt();
    s.nextLine();
    var strs = new String[n];
    for(int i = 0; i < n; i++) strs[i] = s.nextLine();
    var strsSet = new java.util.HashSet<>(java.util.Arrays.asList(strs));
    int resultInt = 0;
    for(int i = n; i < n + m; i++) if(strsSet.contains(s.nextLine())) resultInt++;
    System.out.print(resultInt);
  }
}